### PR TITLE
3765: Conditional Logic: Remove now-unused code

### PR DESCRIPTION
**Pull request by jamesgriff**
@alistairknight 
Before you merge this PR, please:
* checkout `origin/master` on your laptop
* run http://localhost:5001/jobs/migrate-user-input-to-new-format?token=...

This PR removes the now-unused data migration code (and some other unrelated unused code).
I've run the data migration on all environments and checked that the data has migrated successfully.
To check, run `http://ADMIN-HOSTNAME/jobs/check-user-input-formats?token=...`
* My laptop: `0 applications in old format. 2 applications in new format`
* Sandbox: `0 applications in old format. 66 applications in new format`
* Staging: `0 applications in old format. 0 applications in new format`
* Production: `0 applications in old format. 672 applications in new format`

### CODE REVIEW

File: grc/utils/form_custom_validators.py [hunk 1]
```diff
@@ -116,12 +116,6 @@ def validateReferenceNumber(form, field):
         raise ValidationError('Enter a valid reference number')
 
 
-def validateEmailAddress(form, field):
-    from email.utils import parseaddr
-    if not '@' in parseaddr(field.data)[1]:
-        raise ValidationError('Enter a valid email address')
-
-
```
Comment (jamesgriff) [inline grc/utils/form_custom_validators.py:124]: This code isn't used any more.
We seem to be using the `wtforms.validators.Email` validator instead
e.g. https://github.com/cabinetoffice/grc-app/blob/master/admin/admin/forms.py#L10

File: grc/utils/form_custom_validators.py [hunk 2]
```diff
@@ -141,14 +135,6 @@ def password_check(password):
...
 
 
-def validateAdopted(form, field):
-
-    # Custom validator for the "Are you adopted", StrictRequiredIf doesn't play nice with this question
-    if form['check'].data == 'Yes':
-        if field.data is None or field.data == '':
-            raise ValidationError('Select if your were you adopted in the United Kingdom')
-
-
```
Comment (jamesgriff) [inline grc/utils/form_custom_validators.py:151]: This code isn't used any more.
We seem to be using the `wtforms.validators.DataRequired` validator instead
see https://github.com/cabinetoffice/grc-app/blob/master/grc/birth_registration/forms.py#L106

File: grc/utils/reference_number.py [hunk 3]
```diff
@@ -1,23 +1,4 @@
...
-        try:
-            record = Application(reference_number=ref_number, email=email)
-            db.session.add(record)
-            db.session.commit()
-            return ref_number
-        except ValueError:
-            print("Oops!  Something went wrong.")
-    else:
-        print("Reference number exists, trying again")
-        reference_number_generator(email)
```
Comment (jamesgriff) [inline grc/utils/reference_number.py:20]: This has moved to `data_store.py`
https://github.com/cabinetoffice/grc-app/blob/master/grc/business_logic/data_store.py#L72-L79

File: grc/models.py [hunk 4]
```diff
@@ -39,15 +37,7 @@ class Application(db.Model):
...
     def application_data(self) -> ApplicationData:
-        if self.user_input:
-            try:
-                application_data_obj: ApplicationData = jsonpickle.decode(self.user_input)
-            except:
-                application_data_obj = convert_weakly_typed_to_strongly_typed(ast.literal_eval(self.user_input))
-        else:
-            application_data_obj = ApplicationData()
-        
-        return application_data_obj
```
Comment (jamesgriff) [inline grc/models.py:50]: I've run the "Check data migration" endpoint on all our environments.
This checks (for each row in the applications table) if we can successfully run `jsonpickle.decode(application.user_input)`.
There are no rows left (on my laptop, Sandbox, Staging or Production) where this line fails

File: admin/jobs/__init__.py [hunk 5]
```diff
@@ -21,61 +17,6 @@ def index():
...
-
-    db.session.commit()
-
-    return f"Conversion complete." \
-           f" {number_of_applications_in_old_format} converted from old to new format." \
-           f" {number_of_applications_in_new_format} already in new format." \
-           f" Migration failed for {data_migration_failed_for_these_applications}"
-
-
-
```
Comment (jamesgriff) [inline admin/jobs/__init__.py:78]: I've run the migration on all our environments and all rows have been migrated, so we don't need this any more

**Review by alistairknight — APPROVED** [review]

**Pull request merged**